src/main/java/com/gluonhq/substrate/util/ios/CodeSigning.java

```diff
@@ -183,7 +183,7 @@ private MobileProvision tryModifiedBundleId(Identity identity, String bundleId,
 
     private static List<MobileProvision> retrieveValidMobileProvisions() {
         final LocalDate now = LocalDate.now();
-        if (mobileProvision == null) {
+        if (mobileProvisions == null) {
```

Comment:
The reason for the `mobileProvision` check was to allow possible changes in the mobile provisioning folder. If you cache the list, this won't happen.

Reply:
1. do you expect changes on the filesystem, in that folder, during the CodeSigning.signApp() invocation?
2. even if you expect them, it sounds very dangerous to rely on filesystem changes.

Reply:
In theory the signing process will fully stop if it fails the first time because of non valid provisioning profile found in the file system. Any change made to the file system will be checked in the next run.